app/services/katello/pulp3/repository/yum.rb

```diff
@@ -104,14 +107,77 @@ def multi_copy_units(repo_id_map, dependency_solving)
                 data.config << config
               end
             end
-            # FIXME: data's content being [] causes all content to be copied back
-            tasks << api.copy_api.copy_content(data)
+            tasks << copy_pulp_units(data)
           else
             tasks << remove_all_content_from_mapping(repo_id_map)
           end
           tasks.flatten
         end
 
+        def copy_pulp_units(data)
+          tasks = []
+          unit_amount = 0
+          data.config.each { |repo_config| unit_amount += repo_config[:content].size }
+          if unit_amount > UNIT_LIMIT
+            tasks << copy_content_chunked(data)
+          else
+            tasks << api.copy_api.copy_content(data)
+          end
+          tasks
+        end
+
+        def copy_api_data_dup(data)
+          data_dup = PulpRpmClient::Copy.new
+          data_dup.dependency_solving = data.dependency_solving
+          data_dup.config = []
+          data.config.each do |repo_config|
+            config_hash = {
+              source_repo_version: repo_config[:source_repo_version],
+              dest_repo: repo_config[:dest_repo],
+              content: []
+            }
+            config_hash[:dest_base_version] = repo_config[:dest_base_version] if repo_config[:dest_base_version]
+            data_dup.config << config_hash
+          end
+          data_dup
+        end
+
+        def copy_content_chunked(data)
```

Comment:
My test for `copy_content_chunked` is pretty rudimentary.  I'm just checking the number of times that `copy_content` is called and checking the args.  Since I can't use actual content units I'm stubbing quite a bit.

I'll look at adding some more tests here.